src/ray/raylet/worker_pool.cc

```diff
@@ -857,7 +866,10 @@ std::shared_ptr<WorkerInterface> WorkerPool::PopWorker(
       idle_of_all_languages_map_.erase(worker);
       break;
     }
-    if (worker == nullptr) {
+    // If there are jobs exiting, we should re-try later.
+    // Some jobs holding the data will not exit successfully, so we can reuse them
+    // later.
+    if (worker == nullptr && exiting_cnt == 0) {
```

Comment:
Actually I'm a bit unclear on this line. Was there problem where we were starting too many workers?

Reply:
@rkooo567 mentioned that we'd like to wait until the exiting job finished to reduce the number of core workers.

Reply:
Could we not include this change? It may be a good idea in the future, but let's keep this PR scoped to the bug in question.

Reply:
It's a regression without this according to @rkooo567 
@rkooo567  can you give some comments?

Reply:
I think it is okay to include this in a separate PR (since 1.3 object spilling doesn't work perfectly yet anyway now) as long as this passes some worker capping python tests (that says, if it passes CI).

Reply:
`test_multi_tenancy.py`